### PR TITLE
[backend/frontend] Fix ingestion creation (#4020)

### LINKED ISSUE #4020: Ingestion page is broken
Reproduction : 

1. Create a new ingestion on a platform without any ingestion
2. Reload
3. Kaboom

Steps:

1. Create a new attribute current_state_date in Synchronizer
2. Delete by query every synchronizer that have {} as current_state
3. Migrate existing current_state in the new attribute IF DATE FORMAT
4. Change the registrationAttributes into "date"

### PR DESCRIPTION
**Pull request by SouadHadjiat**
### Proposed changes

* migrate current_state field to current_state_date with date type (schema definition)
* fix prepareElementsForIndexing method to better handle objects

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/4020

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: opencti-platform/opencti-graphql/src/migrations/1691677872733-sync-from-current-state-to-current-state-date.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    {
+      script: {
+        source: 'if (ctx._source.current_state == params.emptyCurrentState) {'
+          + 'ctx._source.current_state_date = params.defaultDate'
+          + '} else {'
+          + 'ctx._source.current_state_date = ctx._source.current_state'
+          + '}'
+          + 'ctx._source.remove("current_state")',
+        params: {
+          emptyCurrentState: {},
```
Comment (richard-julien) [inline opencti-platform/opencti-graphql/src/migrations/1691677872733-sync-from-current-state-to-current-state-date.js]: Should not be as string here? '{}'

File: opencti-platform/opencti-graphql/src/migrations/1691677872733-sync-from-current-state-to-current-state-date.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { elUpdateByQueryForMigration } from '../database/engine';
+import { READ_INDEX_INTERNAL_OBJECTS } from '../database/utils';
+import { DatabaseError } from '../config/errors';
+
+export const up = async (next) => {
+  const defaultDate = new Date();
+  defaultDate.setHours(0, 0, 0, 0);
```
Comment (richard-julien) [inline opencti-platform/opencti-graphql/src/migrations/1691677872733-sync-from-current-state-to-current-state-date.js]: Using now() is not sufficient? Why round on hour?

Comment (tomibennett) [reply opencti-platform/opencti-graphql/src/migrations/1691677872733-sync-from-current-state-to-current-state-date.js]: I guess it is, though we replicated this behavior from the creation process for compliance.

File: opencti-platform/opencti-graphql/src/migrations/1691677872733-sync-from-current-state-to-current-state-date.js (hunk 1 above)
Comment (tomibennett) [reply opencti-platform/opencti-graphql/src/migrations/1691677872733-sync-from-current-state-to-current-state-date.js]: We tested it against problematic stored document and the migration executed flawlessly. What benefit do you see to make it a string?

Comment (SouadHadjiat) [reply opencti-platform/opencti-graphql/src/migrations/1691677872733-sync-from-current-state-to-current-state-date.js]: no it's an empty object, not a string

File: opencti-platform/opencti-graphql/src/migrations/1691677872733-sync-from-current-state-to-current-state-date.js (hunk 2 above)
Comment (SouadHadjiat) [reply opencti-platform/opencti-graphql/src/migrations/1691677872733-sync-from-current-state-to-current-state-date.js]: yes we used the same default date as the one used in SyncCreation (`dayStartDate()`)

**Review by SarahBocognano — COMMENTED** [review]
Tested : ✅ Everything works fine for the ingestion page.

I have just notice some minor stuff : 
- The banner message is not tranlasted
- There is no gap between "stream ID" and "running", I think it's not really pretty (even if it was already the case)
- The button "start" and "stop" does not have the right color 

Capture d’écran (1)
Capture d’écran (2)

**Pull request merged**